packages/docs/integrate/integration-guide.md

```diff
@@ -130,19 +130,35 @@ cd packages/api-client
 yarn add axios
 ```
 
+There are two function that can be used to create the client `onCreate` and `init`.
+`onCreate` is executed on every request and can be used to customize the request or settings.
+`init` will be executed once on the midddleware bootsrap and should be use to setup initial GraphQL client instance.
+
 Now in the code editor, open `packages/api-client/src/index.server.ts`. Inside of it, there is the `onCreate` method.
 
 `onCreate` accepts the `settings` parameter, which is a configuration provided in `packages/theme/middleware.config.js`. By default, it's an empty object but can be any configuration you need.
 
 `onCreate` must return an object with at least `config` and `client` properties but it can have any number of custom properties if needed. This object is later available in API endpoints.
 
-Let's update the `onCreate` method to create and return a new Axios instance.
+
+Let's update the `onCreate` method to simply return `config` and `client` and to execute the `init` function in case that client was not initialized.
+
+```typescript
+// packages/api-client/index.server.ts
+if (!settings?.client) {
+    return init(settings);
+  }
+
+return { config: settings, client: settings.client };
+```
+
+Now let's implement `init` function that will create the GraphQL client. In fact, init function can return any result that will be merged into the configuration object. But for our purpose we will simply create the `axios` client instance.
```

Comment:
```suggestion
Now let's implement the `init` function. In fact, it can return any result, and the server will merge it into the configuration object. But for our purpose, we will create the `axios` client instance.
```